src/evict/evict_lru.c

```diff
@@ -993,14 +1015,21 @@ __evict_lru_walk(WT_SESSION_IMPL *session)
 			    candidates < entries / 2;
 			    candidates++)
 				if (__evict_read_gen(
-				    &cache->evict_queue[candidates]) > cutoff)
+				    &evict_queue->evict_queue[candidates]) >
+				    cutoff)
 					break;
-			cache->evict_candidates = candidates;
+			evict_queue->evict_candidates = candidates;
 		}
 	}
 
-	cache->evict_current = cache->evict_queue;
-	__wt_spin_unlock(session, &cache->evict_lock);
+	__wt_spin_unlock(session, &evict_queue->evict_lock);
+	/*
+	 * Now we can set the next queue.
+	 */
```

Comment:
I think it would be interesting to add some statistics here. One that tracks that we switched the queue. Possibly also information about the state of the queues (e.g: if the one being switched out is empty) - though there is a tradeoff between accuracy and doing more work while holding the ``evict_queue_lock``.